creator-node/src/services/stateMachineManager/stateReconciliation/recoverOrphanedData.jobProcessor.ts

```diff
@@ -18,40 +31,178 @@ const { QUEUE_NAMES } = require('../stateMachineConstants')
  * @param {Object} param.logger the logger that can be filtered by jobName and jobId
  */
 module.exports = async function ({
+  discoveryNodeEndpoint,
   logger
 }: DecoratedJobParams<RecoverOrphanedDataJobParams>): Promise<
   DecoratedJobReturnValue<RecoverOrphanedDataJobReturnValue>
 > {
-  // TODO: Clear redis state
-
-  // TODO: Query this node's db to find all users who have data on it and save them in a redis set
-  const usersOnThisNode = {}
-
-  /**
-   * TODO: Query Discovery to find all users who have this node as their primary or secondary.
-   *       This will do some kind of while loop that requests paginated responses from
-   *       `/v1/full/users/content_node/all?creator_node_endpoint=<THIS_CONTENT_NODE>` and adds
-   *       them to a redis set until reaching the end of the pagination
-   */
-  const usersWithThisNodeInReplica = {}
-
-  // TODO: Perform a set difference between usersOnThisNode and usersWithThisNodeInReplica.
-  //       https://redis.io/commands/sdiff/
-  const usersWithOrphanedData: any[] = []
-
-  /**
-   *  TODO: Run modified "force wipe" (modified forceResync=true flag) on each user with orphaned data:
-   *        1. Merge orphaned data from this node into the user's primary.
-   *        2. Wipe the user's data from this node.
-   *        3. *DON'T* resync from the primary to this node -- this is what non-modified forceResync would do.
-   * */
-
-  // Enqueue another job to search for any new data that gets orphaned after this job finishes
-  const jobsToEnqueue: JobsToEnqueue = {
-    [QUEUE_NAMES.RECOVER_ORPHANED_DATA]: [{}]
+  const numWalletsOnNode = await _storeWalletsOnThisNode()
+  const numUsersWithNodeInReplicaSet = await _storeWalletsWithThisNodeInReplica(
+    discoveryNodeEndpoint
+  )
+
+  // Wallets with orphaned data = set difference between walletsOnThisNode and walletsWithThisNodeInReplicaSet
+  const walletsWithOrphanedOrUnsyncedData: string[] = await redisClient.sdiff(
+    WALLETS_ON_NODE_KEY,
+    WALLETS_WITH_NODE_IN_REPLICA_SET_KEY
+  )
+
+  // TODO: Filter wallets that had this node in RS but didn't have data on this node (these are just unsynced users).
+  //       Could use SSCAN but that's O(N). Alternatively, when enqueuing the sync we could check which set the wallet is in and not wipe if the node is in the user's replica set
```

Comment:
this is a detail that I'm not sure came up before but want to get your opinion @SidSethi @dmanjunath -- so with our 2 sets:

set A: users with data on node
set B: users with node in replica set

set diff includes elements in A not B, which is what we want (users with data on node but not node in replica set) BUT it also includes elements that are in B not A (users with node in replica set but no data on node). I believe this is the same as unsynced users, so we probably wouldn't want to wipe their state. does it sound reasonable to filter these users out via an expensive redis query?

Reply:
yeah fine to filter out via query imo, lets discuss if this is best approach tho

Reply:
actually thought of a more elegant way to do this, which I'm adding now:
```js
SDIFFSTORE A B C // Put set diff into new set, C
SUNION A C // Filter out elements that only belong to set B
```